vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -333,9 +333,20 @@ def __init__(self, vllm_config: VllmConfig, engine_id: str):
         # Metadata.
         self.engine_id = engine_id
         self.rank = get_tensor_model_parallel_rank()
+
+        logger.debug("NIXL worker %s TP rank %s", self.engine_id, self.rank)
         self.world_size = get_tensor_model_parallel_world_size()
         self.tp_group = get_tp_group()
 
+        # NOTE(weaton): Hack to get rank uniqueness across TP + DP ranks.
+        # Used for NIXL side channel communication.
+        self.unique_rank = (self.rank *
+                            self.world_size) + get_dp_group().rank_in_group
```

Comment:
@wseaton I think this should be switched?
```suggestion
        self.unique_rank = (get_dp_group().rank_in_group *
                            self.world_size) + self.rank
```